src/IndexRegistry.sol

```diff
@@ -117,121 +126,170 @@ contract IndexRegistry is IndexRegistryStorage {
     *******************************************************************************/
 
     /**
-     * @notice updates the total numbers of operator in `quorumNumber` to `numOperators`
-     * @param quorumNumber is the number of the quorum to update
-     * @param numOperators is the number of operators in the quorum
+     * @notice Increases the historical operator count by 1 and returns the new count
      */
-    function _updateTotalOperatorHistory(uint8 quorumNumber, uint32 numOperators) internal {
-        _totalOperatorsHistory[quorumNumber].push(QuorumUpdate({
-            numOperators: numOperators,
-            fromBlockNumber: uint32(block.number)
-        }));
+    function _increaseOperatorCount(uint8 quorumNumber) internal returns (uint32) {
+        QuorumUpdate storage lastUpdate = _latestQuorumUpdate(quorumNumber);
```

Comment:
seems unrelated to this PR(?) but I think calling this struct 'QuorumUpdate' is super vague. It should probably be an 'OperatorCountUpdate' or something similar